backend/spec/integration/restaurants_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'swagger_helper'
+
+describe 'Restaurants API' do
```

Comment:
Notice you're not testing the entire filtering and paging logic - for a good reason. It's very hard to test something that does so many things (another reason to break large classes into smaller ones)

Reply:
I  might have missed it but I see no reviews average calculation....